packages/web3-providers-ws/src/index.js

```diff
@@ -207,7 +247,11 @@ WebsocketProvider.prototype._addSocketListeners = function () {
     this.connection.addEventListener('message', this._onMessage.bind(this));
     this.connection.addEventListener('open', this._onConnect.bind(this));
     this.connection.addEventListener('close', this._onClose.bind(this));
-};
+    if(this.connection._connection){
+        this.connection._client.removeAllListeners('connectFailed'); //Override the internal listeners, so they don't trigger a `close` event. We want to trigger `_onClose` manually with a description.
```

Comment:
As browsers uses the native [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) object + doesn't use/have ws lib client and node uses ws client. so can we more specifically check `_client` here?

Reply:
And do we also need to handle the same for the browser WebSocket? Or is it only an issue that we need to handle just for node?